corehq/apps/userreports/reports/data_source.py

```diff
@@ -123,22 +122,38 @@ def get_data(self, slugs=None):
             if self._order_by:
                 for col in reversed(self._order_by):
                     is_descending = col[1] == DESCENDING
-                    is_date = any(
-                        configured_indicator['datatype'] == 'date'
-                        for configured_indicator in self.config.configured_indicators
-                        if configured_indicator['column_id'] == col[0]
-                    )
-                    default_sort_by_date = (
-                        date(datetime.MINYEAR, 1, 1)
-                        if is_descending else date(datetime.MAXYEAR, 12, 31)
-                    )
-                    value = lambda x: x.get(col[0], None)
-                    sort_by_value = lambda x: (
-                        value(x)
-                        or (default_sort_by_date if is_date else value(x))
-                    )
+
+                    def sort_by(row):
+                        value = row.get(col[0], None)
+                        if not value:
+                            matching_indicators = filter(
+                                lambda configured_indicator: configured_indicator['column_id'] == col[0],
```

Comment:
if that doesn't work then I think you either have to look up the type from the field or you have to trap the error and disallow sorting. i think it could also be a problem if the first column is a compound column like a percentage though maybe i don't understand exactly what this is doing